helmclient.go

```diff
@@ -693,12 +720,68 @@ func getTillerImage(pod *corev1.Pod) (string, error) {
 
 	tillerImage := pod.Spec.Containers[0].Image
 	if tillerImage == "" {
-		return "", microerror.Mask(tillerImageNotFoundError)
+		return "", microerror.Mask(tillerImageInvalidError)
 	}
 
 	return tillerImage, nil
 }
 
+func isTillerOutdated(pod *corev1.Pod) error {
+	currentTillerImage, err := getTillerImage(pod)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	currentTillerVersion, err := parseTillerVersion(currentTillerImage)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	desiredTillerVersion, err := parseTillerVersion(tillerImageSpec)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	for i, currentVersion := range currentTillerVersion {
+		desiredVersion := desiredTillerVersion[i]
+		if currentVersion < desiredVersion {
+			return microerror.Maskf(tillerOutdatedError, "%#q older than %#q", currentTillerVersion, desiredTillerVersion)
+		}
+	}
+
+	return nil
+}
+
+func parseTillerVersion(tillerImage string) ([]int, error) {
+	version := make([]int, 3)
+
+	// Tiller image tag has the version.
+	imageParts := strings.Split(tillerImage, ":v")
+	if len(imageParts) != 2 {
+		return version, microerror.Maskf(tillerImageInvalidError, "%s", tillerImage)
+	}
+
+	// Version may be a release candidate. If so remove the -rc suffix.
+	tag := imageParts[1]
+	tagParts := strings.Split(tag, "-")
+
+	versionParts := strings.Split(tagParts[0], ".")
+	if len(versionParts) != 3 {
+		return version, microerror.Maskf(tillerImageInvalidError, "%s", tillerImage)
+	}
+
+	for i, s := range versionParts {
+		v, err := strconv.Atoi(s)
+		if err != nil {
+			return version, microerror.Mask(err)
```

Comment:
IMO ere it would be good to have an error saying something like `"can not convert %d part of version %#q", i, tag`.